### PR TITLE
Add mizar network policy json conversion

### PR DESCRIPTION
**Pull request by clu2xlu**
**What type of PR is this?**

Mizar and Arktos Integration for Network Policy feature

/kind feature

**What this PR does / why we need it**:
Mizar network policy communication is through grpc. Mizar is using Python as the programing language and Arktos is using Go. 
In order to have Mizar to read network policy metadata, we need convert Go presentation into a json string and pass it to mizar through GRPC. 
This PR is for json string conversion of the network policy

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by Hong-Chang — APPROVED** [review]
LGTM

File: pkg/controller/mizar/util.go [hunk 1]
```diff
@@ -41,6 +42,47 @@ type KeyWithEventType struct {
...
+}
+
+type MizarNetworkPolicyIngressMsg struct {
+	Ports []MizarNetworkPolicyPortSelector `json:"ports"`
+	Rules []MizarNetworkPolicyRule         `json:"from"`
+}
+
+type MizarNetworkPolicyEgressMsg struct {
+	Ports []MizarNetworkPolicyPortSelector `json:"ports"`
+	Rules []MizarNetworkPolicyRule         `json:"to"`
```
Comment (chenqianfzh) [inline pkg/controller/mizar/util.go:76]: json name is "to", yet the variable nmae is "Rules".

File: pkg/controller/mizar/util.go [hunk 2]
```diff
@@ -85,7 +127,7 @@ func ConvertToPodContract(pod *v1.Pod) *BuiltinsPodMessage {
 		network = ""
 	}
 
-	klog.V(3).Infof("Pod Name: %s, HostIP: %s, Namespace: %s, Tenant: %s, Arktos network: %s",
+	klog.Infof("Pod Name: %s, HostIP: %s, Namespace: %s, Tenant: %s, Arktos network: %s",
```
Comment (chenqianfzh) [inline pkg/controller/mizar/util.go:130]: network: %s"

Is network of string type? If not, "%v" is a better formatter.

File: pkg/controller/mizar/util.go [hunk 3]
```diff
@@ -107,20 +149,213 @@ func ConvertToNodeContract(node *v1.Node) *BuiltinsNodeMessage {
...
 
 	return &BuiltinsNetworkPolicyMessage{
 		Name:      policy.Name,
 		Namespace: policy.Namespace,
 		Tenant:    policy.Tenant,
+		Policy:    string(policyJson),
+	}
+}
+
+func parseNetworkPolicySpecToMsg(nps networking.NetworkPolicySpec) MizarNetworkPolicyPolicySpecMsg {
```
Comment (chenqianfzh) [inline pkg/controller/mizar/util.go:173]: This function can be simplifed as below ( most local variables can be skipped):
{
	return MizarNetworkPolicyPolicySpecMsg{
		PodSel: MizarNetworkPolicyPodSelector{
		     MatchLabels: nps.PodSelector.MatchLabels,
	        }
		In:     parseNetworkPolicyIngressRulesToMsg(nps.Ingress),
		Out:    parseNetworkPolicyEgressRulesToMsg(nps.Egress),
		Type:   policyTypesToStringArray(nps.PolicyTypes),
	}
}

File: pkg/controller/mizar/util.go [hunk 4]
```diff
@@ -63,7 +105,7 @@ func ConvertToServiceEndpointContract(endpoints *v1.Endpoints, service *v1.Servi
 	}
 	portsJson, _ := json.Marshal(ports)
 
-	klog.V(3).Infof("Endpoint Name: %s, Namespace: %s, Tenant: %s, Backend Ips: %s, Ports: %s",
+	klog.Infof("Endpoint Name: %s, Namespace: %s, Tenant: %s, Backend Ips: %s, Ports: %s",
```
Comment (vinaykul) [inline pkg/controller/mizar/util.go:108]: Why did we change the log-level?

File: pkg/controller/mizar/util.go [hunk 5]
```diff
@@ -107,20 +149,213 @@ func ConvertToNodeContract(node *v1.Node) *BuiltinsNodeMessage {
 		}
 	}
 
-	klog.V(3).Infof("Node Name: %s, IP: %s", node.Name, ip)
+	klog.Infof("Node Name: %s, IP: %s", node.Name, ip)
```
Comment (vinaykul) [inline pkg/controller/mizar/util.go:152]: Same here for log level

**Review by vinaykul — CHANGES_REQUESTED** [review]
please review the log levels . In order to avoid cluttering up the logs, please keep them at V(3) unless it is unavoidable.

**Review by clu2xlu — COMMENTED** [review]
In case of 
```
from.PodSelector == nil && from.NamespaceSelector == nil && from.IPBlock == nil 
```
it should return empty. 

```
{"podSelector":{},"policyTypes":["Ingress"]}
```
This test case is already addressed

**Pull request merged**